checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -259,6 +259,9 @@ protected void commonAssignmentCheck(
         // Use the valueExp as the context because data flow will have a value for that tree.  It
         // might not have a value for the var tree.  This is sound because if data flow has
         // determined @PolyNull is @Nullable at the RHS, then it is also @Nullable for the LHS.
+        // TODO: replacePolyQualifier is a temporary hack as it produce side effect to polymorphic
```

Comment:
How about: `A visitor should not change types, so this call to replacePolyQualifier is a hack. To work around this hack here, NullnessVisitor#visitConditionalExpression needs a further hack. Both should be undone, by properly resolving polymorphic qualifiers in the ATF/transfer.`

Reply:
Cool, thanks. I definitely should try to express more clearly.